src/phoenix/trace/llama_index/callback.py

```diff
@@ -308,44 +313,47 @@ def _add_spans_to_tracer(
                 if preceding_event_parent_span_id != parent_span_id:
                     break
                 preceding_event_data = event_id_to_event_data[preceding_event_id]
-                if preceding_event_data["event_type"] is not CBEventType.TEMPLATING:
+                if preceding_event_data.event_type is not CBEventType.TEMPLATING:
                     break
                 parent_child_id_stack.pop()
-                if payload := preceding_event_data["start_event"].payload:
-                    # Add template attributes to the LLM span to which they belong.
-                    attributes.update(_template_attributes(payload))
-
-        start_event = event_data["start_event"]
-        start_time = _timestamp_to_tz_aware_datetime(start_event.time)
-        if event_type is CBEventType.EXCEPTION:
-            # LlamaIndex has exception callback events that are sibling events of the events in
-            # which the exception occurred. We collect all the exception events and add them to the
-            # relevant span.
-            if (
-                not start_event.payload
-                or (error := start_event.payload.get(EventPayload.EXCEPTION)) is None
-            ):
-                continue
-            span_exceptions.append(
-                SpanException(
-                    message=str(error),
-                    timestamp=start_time,
-                    exception_type=type(error).__name__,
-                    exception_stacktrace=get_stacktrace(error),
+                if preceding_event_start := preceding_event_data.start_event:
+                    if preceding_payload := preceding_event_start.payload:
+                        # Add template attributes to the LLM span to which they belong.
+                        attributes.update(_template_attributes(preceding_payload))
+
+        start_time = None
+        if start_event := event_data.start_event:
+            start_time = _timestamp_to_tz_aware_datetime(start_event.time)
+            if event_type is CBEventType.EXCEPTION:
+                # LlamaIndex has exception callback events that are sibling events of the events in
+                # which the exception occurred. We collect all the exception events and add them to
+                # the relevant span.
+                if (
+                    not start_event.payload
+                    or (error := start_event.payload.get(EventPayload.EXCEPTION)) is None
+                ):
+                    continue
+                span_exceptions.append(
+                    SpanException(
+                        message=str(error),
+                        timestamp=start_time,
+                        exception_type=type(error).__name__,
+                        exception_stacktrace=get_stacktrace(error),
+                    )
                 )
-            )
-            continue
+                continue
```

Comment:
The `continue` is actually needed so the rest of the iteration is skipped when the event type is exception, so it can't be nested inside the `if` statement. This also means `start_time` needs a value beforehand.

Reply:
got it, thanks!